### PR TITLE
Migrate permalink view into redux-style and remove jquery

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
While working on [MM-9708](https://mattermost.atlassian.net/browse/MM-9708), I bumped into migrating this PermalinkView component.

#### Ticket Link
none

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

File: components/permalink_view/permalink_view.jsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+    doPermalinkEvent = async (props) => {
+        this.setState({valid: false});
+        const postId = props.match.params.postid;
+        await emitPostFocusEvent(postId, this.props.returnTo);
+        this.setState({valid: true});
+    }
+
+    isStateValid = () => {
+        return this.state.valid && this.props.channelId !== '' && this.props.teamName;
```
Comment (enahum) [inline components/permalink_view/permalink_view.jsx]: I think the `!== ''` is not actually necessary... non-blocking

File: components/permalink_view/permalink_view.jsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+    render() {
+        const {
+            channelId,
+            channelName,
+            teamName,
+        } = this.props;
+
+        const postId = this.props.match.params.postid;
```
Comment (enahum) [inline components/permalink_view/permalink_view.jsx]: can you add `match` to the propTypes ?

**Comment by saturninoabril** [comment]
@enahum updated per your comment. thanks!

File: components/permalink_view/permalink_view.jsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+    isStateValid = () => {
+        return this.state.valid && this.props.channelId && this.props.teamName;
+    }
+
+    render() {
+        const {
+            channelId,
+            channelName,
+            match,
```
Comment (enahum) [inline components/permalink_view/permalink_view.jsx]: I really don't understand how this is not throwing an ESLint error, what I meant is that you should define `match` in the above
```js
static propTypes = {
...
}
```

Comment (saturninoabril) [reply components/permalink_view/permalink_view.jsx]: Ah I see, and also we have similar case in several components as well. I'll submit separate PR to define `match` to other components.

Comment (saturninoabril) [reply components/permalink_view/permalink_view.jsx]: `match` is ignored in our `.eslintrc.json`,
```
"react/prop-types": [2, { "ignore": ["match", "location", "history", "component"] }],
```

Comment (enahum) [reply components/permalink_view/permalink_view.jsx]: hmm, interesting.. but yes I do prefer to have it explicitly

**Comment by koxen** [comment]
document.body.classList.add('app__body'); 

can we get rid of app__body hacks like this? is it really needed?

**Comment by saturninoabril** [comment]
Primary intent is to get rid off jQuery. If there's better alternative, feel free to submit and we're very much happy to have it :)

**Pull request merged**